WcaOnRails/app/models/user.rb

```diff
@@ -327,6 +327,34 @@ def avatar_requires_wca_id
     end
   end
 
+  after_save :changed_delegate_status
+  private def changed_delegate_status
+    user = User.find_by_wca_id(wca_id)
+    if saved_change_to_delegate_status? && !delegate_status
+      senior_delegate = User.find_by_id(senior_delegate_id_before_last_save)
+      DelegateStatusChangeMailer.notify_board_and_wqac_of_delegate_status_change(user, senior_delegate).deliver_later
+    elsif saved_change_to_delegate_status? && delegate_status
+      senior_delegate = user.senior_delegate
+      DelegateStatusChangeMailer.notify_board_and_wqac_of_delegate_status_change(user, senior_delegate).deliver_later
+    end
+  end
+
+  def delegate_status_name
```

Comment:
This sort of stuff does not belong in a model. We also already have this mapping in our translation files. See https://github.com/jfly/worldcubeassociation.org/blob/efcd848621418fd54dd6d7c8be42faf2f2dd1511/WcaOnRails/config/locales/en.yml#L312-L315. We actually use the enum_help gem to help out with this stuff (see https://github.com/zmbacker/enum_help#usage). You should be able to convert a user to a pretty string of their delegate status by doing `user.delegate_status_i18n`

Reply:
Oh, I see. I remembered about [this](https://github.com/thewca/worldcubeassociation.org/pull/2040/files#diff-549c86482e89e487f701679812eac11e) and went straight ahead with it. Thanks for the neat fix :D

Reply:
If, for instance, I use the website in Spanish and make a change, the automatic email will have "Delegado" instead of "Delegate" because of the usage of user.delegate_status_i18n? I'm seeing this on the preview, and I think that the email should be fully in English.

Reply:
Oh, good point! The way we've handled this in the past is to wrap emails that should be in English with a `I18n.with_locale :en ...`. See [notify_of_delegate_report_submission](https://github.com/jfly/worldcubeassociation.org/blob/da4883f4d9facbfef13458ff23b559347a47a010/WcaOnRails/app/mailers/competitions_mailer.rb#L38-L48) for an example of this.